src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -0,0 +1,160 @@
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine.Direction;
+import java.util.function.Supplier;
+import monologue.Annotations.Log;
+import monologue.Logged;
+import org.sciborgs1155.robot.Constants;
+import org.sciborgs1155.robot.Robot;
+
+public class Pivot extends SubsystemBase implements AutoCloseable, Logged {
+  @Log.NT private final PivotIO pivot;
+  private final SysIdRoutine sysIdRoutine; //sysIdRoutineoogabooga
+
+  // pivot control
+  @Log.NT 
+  final ProfiledPIDController pivotPID =
+      new ProfiledPIDController(
+          PivotConstants.kP,
+          PivotConstants.kI,
+          PivotConstants.kD,
+          new TrapezoidProfile.Constraints(PivotConstants.MAX_VELOCITY, PivotConstants.MAX_ACCEL));
+  @Log.NT private final ArmFeedforward pivotFeedforward =
+      new ArmFeedforward(PivotConstants.kS, PivotConstants.kG, PivotConstants.kV);
+
+  // climb control
+  @Log.NT private final ProfiledPIDController climbPID =
+      new ProfiledPIDController(
+          ClimbConstants.kP,
+          ClimbConstants.kI,
+          ClimbConstants.kD,
+          new TrapezoidProfile.Constraints(ClimbConstants.MAX_VELOCITY, ClimbConstants.MAX_ACCEL));
+  @Log.NT private final ArmFeedforward climbFeedforward =
+      new ArmFeedforward(ClimbConstants.kS, ClimbConstants.kG, ClimbConstants.kV);
+
+  // visualization
+  @Log.NT final Mechanism2d measurement = new Mechanism2d(3, 4);
+  @Log.NT final Mechanism2d setpoint = new Mechanism2d(3, 4);
+
+  private final PivotVisualizer positionVisualizer =
+      new PivotVisualizer(measurement, new Color8Bit(255, 0, 0));
+  private final PivotVisualizer setpointVisualizer =
+      new PivotVisualizer(setpoint, new Color8Bit(0, 0, 255));
+
+  /** Creates a real or simulated pivot based on {@link Robot#isReal()} */
+  public static Pivot create() {
+    return Robot.isReal() ? new Pivot(new RealPivot()) : new Pivot(new SimPivot());
+  }
+
+  public Pivot(PivotIO pivot) {
+    this.pivot = pivot;
+    sysIdRoutine =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(),
+            new SysIdRoutine.Mechanism(
+                (v) -> pivot.setVoltage(v.in(Volts)), null, this, "pivotSysId"));
+
+    SmartDashboard.putData("quasistaticForward", quasistaticForward());
+    SmartDashboard.putData("quasistaticBack", quasistaticBack());
+    SmartDashboard.putData("dynamicForward", dynamicForward());
+    SmartDashboard.putData("dynamicBack", dynamicBack());
+  }
+
+  /**
+   * Smoothly angle the pivot to a desired goal using a {@link ProfiledPIDController}.
+   *
+   * @param goalAngle The position to move the pivot to.
+   * @return The command to set the pivot's angle.
+   */
+  public Command runPivot(Supplier<Rotation2d> goalAngle) {
+    return runOnce(() -> pivotPID.setGoal(goalAngle.get().getRadians()))
```

Comment:
this should be one command, not a sequential command
the pivotPID.calculate should take in both measurement and goal.get()